src/python/pants/backend/python/tasks2/resolve_requirements_task_base.py

```diff
@@ -47,10 +55,16 @@ def resolve_requirements(self, req_libs):
       # to cover the empty case.
       if not os.path.isdir(path):
         with safe_concurrent_creation(path) as safe_path:
-          self._build_requirements_pex(interpreter, safe_path, req_libs)
+          self._build_requirements_pex(interpreter, safe_path, req_libs,
+                                       local_python_dist_targets=local_python_dist_targets)
     return PEX(path, interpreter=interpreter)
 
-  def _build_requirements_pex(self, interpreter, path, req_libs):
+  def _build_requirements_pex(self, interpreter, path, req_libs, local_python_dist_targets=None):
     builder = PEXBuilder(path=path, interpreter=interpreter, copy=True)
     dump_requirements(builder, interpreter, req_libs, self.context.log)
+    # Dump built python distributions, if any, into requirements pex.
+    built_dists = self.context.products.get_data(PythonCreateDistributions.PYTHON_DISTS)
+    if built_dists:
+      for dist in built_dists:
+        builder.add_dist_location(dist)
```

Comment:
seems to me like this is sidestepping the actual resolve, but ultimately I'm curious if this covers the case of:

1) user has a target that depends on `3rdparty/python:pytorch` which provides e.g. `pytorch==1.0.0`
2) user also depends on a `python_dist` that provides `pytorch==2.0.0`
3) when the pex is built, it blows up indicating two incompatible versions of pytorch?

if so, great. if not, this could spell out trouble at runtime - particularly because of transitive deps that users might not be aware of. either way this case should probably get covered with an integration test.

Reply:
Great question. Let us get back to you on that, definitely need an integ test regardless.

Reply:
that's a really good observation.

Reply:
Does this issue get addressed here?

Reply:
This is covered in the same way a resolve would blow up on normal python targets (i.e. throwing an error indicating two incompatible versions of pytorch). I have an integration test covering this.